Comparing buffers hangs when computing the diff report

I am having the very same issue as #1433 mocha 2.2.1 and chai 2.1.1. When 2 buffers are not equal, the diff computation hangs and CPU gets crazy.

Any chance of a regression bug here?

Sample code to reproduce this:

```js
var chai = require('chai');
var expect = chai.expect;

// chai.config.showDiff = false; // Without this line, the diff generation hangs

describe('Testing buffer equality', function () {
  it('should not take foreveeeeeer', function () {
    var file1 = Fs.readFileSync('/tmp/file1');
    var file2 = Fs.readFileSync('/tmp/file2');
    expect(file1).to.deep.equal(file2);
  });
});
```

At this point of writing, I am unclear if this issue comes from mocha or from chai, though...

I tested right now, and it work pretty good. here's the output:
```sh
 Testing buffer equality
    1) should not take foreveeeeeer


  0 passing (6ms)
  1 failing

  1) Testing buffer equality should not take foreveeeeeer:

      AssertionError: expected <Buffer 66 6f 6f 0a> to deeply equal <Buffer 62 61 72 0a>
      + expected - actual

       [
      +  98
      +  97
      +  114
      -  102
      -  111
      -  111
         10
       ]
```
maybe you deal with a large files @astorije ?

Confirmed on the reproduction locally for specifically the `Shared-reference` hang case. It hangs for me.

The other two don't. Commenting out the shared-reference hang block, here's an abridged output copy & paste:

```plaintext
$ npx mocha repro.test.js

  Buffer diff hang
    1) should not hang comparing 20KB+ buffers

  Wide object hang
    2) should not hang on objects with thousands of keys


  0 passing (36ms)
  2 failing

  1) Buffer diff hang
       should not hang comparing 20KB+ buffers:

      AssertionError: expected Buffer[ 65, 65, 65, 65, 65, 65, …(20416) ] to deeply equal Buffer[ 66, 66, 66, 66, 66, 66, …(20415) ]
      + expected - actual

      ...

      [mocha] output truncated to 8192 characters, see "maxDiffSize" reporter-option

      at Context.<anonymous> (repro.test.js:10:26)
      at process.processImmediate (node:internal/timers:504:21)

  2) Wide object hang
       should not hang on objects with thousands of keys:

      AssertionError: expected { Object (k0, k1, ...) } to deeply equal {}
      + expected - actual

      ...

      [mocha] output truncated to 8192 characters, see "maxDiffSize" reporter-option

      at Context.<anonymous> (repro.test.js:32:20)
      at process.processImmediate (node:internal/timers:504:21)
```

<details>
<summary>Full test output here</summary>

```plaintext
$ npx mocha repro.test.js

  Buffer diff hang
    1) should not hang comparing 20KB+ buffers

  Wide object hang
    2) should not hang on objects with thousands of keys


  0 passing (36ms)
  2 failing

  1) Buffer diff hang
       should not hang comparing 20KB+ buffers:

      AssertionError: expected Buffer[ 65, 65, 65, 65, 65, 65, …(20416) ] to deeply equal Buffer[ 66, 66, 66, 66, 66, 66, …(20415) ]
      + expected - actual

       [
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      -  65
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      +  66
      
      [mocha] output truncated to 8192 characters, see "maxDiffSize" reporter-option

      at Context.<anonymous> (repro.test.js:10:26)
      at process.processImmediate (node:internal/timers:504:21)

  2) Wide object hang
       should not hang on objects with thousands of keys:

      AssertionError: expected { Object (k0, k1, ...) } to deeply equal {}
      + expected - actual

      -{
      -  "k0": 0
      -  "k1": 1
      -  "k10": 10
      -  "k100": 100
      -  "k1000": 1000
      -  "k1001": 1001
      -  "k1002": 1002
      -  "k1003": 1003
      -  "k1004": 1004
      -  "k1005": 1005
      -  "k1006": 1006
      -  "k1007": 1007
      -  "k1008": 1008
      -  "k1009": 1009
      -  "k101": 101
      -  "k1010": 1010
      -  "k1011": 1011
      -  "k1012": 1012
      -  "k1013": 1013
      -  "k1014": 1014
      -  "k1015": 1015
      -  "k1016": 1016
      -  "k1017": 1017
      -  "k1018": 1018
      -  "k1019": 1019
      -  "k102": 102
      -  "k1020": 1020
      -  "k1021": 1021
      -  "k1022": 1022
      -  "k1023": 1023
      -  "k1024": 1024
      -  "k1025": 1025
      -  "k1026": 1026
      -  "k1027": 1027
      -  "k1028": 1028
      -  "k1029": 1029
      -  "k103": 103
      -  "k1030": 1030
      -  "k1031": 1031
      -  "k1032": 1032
      -  "k1033": 1033
      -  "k1034": 1034
      -  "k1035": 1035
      -  "k1036": 1036
      -  "k1037": 1037
      -  "k1038": 1038
      -  "k1039": 1039
      -  "k104": 104
      -  "k1040": 1040
      -  "k1041": 1041
      -  "k1042": 1042
      -  "k1043": 1043
      -  "k1044": 1044
      -  "k1045": 1045
      -  "k1046": 1046
      -  "k1047": 1047
      -  "k1048": 1048
      -  "k1049": 1049
      -  "k105": 105
      -  "k1050": 1050
      -  "k1051": 1051
      -  "k1052": 1052
      -  "k1053": 1053
      -  "k1054": 1054
      -  "k1055": 1055
      -  "k1056": 1056
      -  "k1057": 1057
      -  "k1058": 1058
      -  "k1059": 1059
      -  "k106": 106
      -  "k1060": 1060
      -  "k1061": 1061
      -  "k1062": 1062
      -  "k1063": 1063
      -  "k1064": 1064
      -  "k1065": 1065
      -  "k1066": 1066
      -  "k1067": 1067
      -  "k1068": 1068
      -  "k1069": 1069
      -  "k107": 107
      -  "k1070": 1070
      -  "k1071": 1071
      -  "k1072": 1072
      -  "k1073": 1073
      -  "k1074": 1074
      -  "k1075": 1075
      -  "k1076": 1076
      -  "k1077": 1077
      -  "k1078": 1078
      -  "k1079": 1079
      -  "k108": 108
      -  "k1080": 1080
      -  "k1081": 1081
      -  "k1082": 1082
      -  "k1083": 1083
      -  "k1084": 1084
      -  "k1085": 1085
      -  "k1086": 1086
      -  "k1087": 1087
      -  "k1088": 1088
      -  "k1089": 1089
      -  "k109": 109
      -  "k1090": 1090
      -  "k1091": 1091
      -  "k1092": 1092
      -  "k1093": 1093
      -  "k1094": 1094
      -  "k1095": 1095
      -  "k1096": 1096
      -  "k1097": 1097
      -  "k1098": 1098
      -  "k1099": 1099
      -  "k11": 11
      -  "k110": 110
      -  "k1100": 1100
      -  "k1101": 1101
      -  "k1102": 1102
      -  "k1103": 1103
      -  "k1104": 1104
      -  "k1105": 1105
      -  "k1106": 1106
      -  "k1107": 1107
      -  "k1108": 1108
      -  "k1109": 1109
      -  "k111": 111
      -  "k1110": 1110
      -  "k1111": 1111
      -  "k1112": 1112
      -  "k1113": 1113
      -  "k1114": 1114
      -  "k1115": 1115
      -  "k1116": 1116
      -  "k1117": 1117
      -  "k1118": 1118
      -  "k1119": 1119
      -  "k112": 112
      -  "k1120": 1120
      -  "k1121": 1121
      -  "k1122": 1122
      -  "k1123": 1123
      -  "k1124": 1124
      -  "k1125": 1125
      -  "k1126": 1126
      -  "k1127": 1127
      -  "k1128": 1128
      -  "k1129": 1129
      -  "k113": 113
      -  "k1130": 1130
      -  "k1131": 1131
      -  "k1132": 1132
      -  "k1133": 1133
      -  "k1134": 1134
      -  "k1135": 1135
      -  "k1136": 1136
      -  "k1137": 1137
      -  "k1138": 1138
      -  "k1139": 1139
      -  "k114": 114
      -  "k1140": 1140
      -  "k1141": 1141
      -  "k1142": 1142
      -  "k1143": 1143
      -  "k1144": 1144
      -  "k1145": 1145
      -  "k1146": 1146
      -  "k1147": 1147
      -  "k1148": 1148
      -  "k1149": 1149
      -  "k115": 115
      -  "k1150": 1150
      -  "k1151": 1151
      -  "k1152": 1152
      -  "k1153": 1153
      -  "k1154": 1154
      -  "k1155": 1155
      -  "k1156": 1156
      -  "k1157": 1157
      -  "k1158": 1158
      -  "k1159": 1159
      -  "k116": 116
      -  "k1160": 1160
      -  "k1161": 1161
      -  "k1162": 1162
      -  "k1163": 1163
      -  "k1164": 1164
      -  "k1165": 1165
      -  "k1166": 1166
      -  "k1167": 1167
      -  "k1168": 1168
      -  "k1169": 1169
      -  "k117": 117
      -  "k1170": 1170
      -  "k1171": 1171
      -  "k1172": 1172
      -  "k1173": 1173
      -  "k1174": 1174
      -  "k1175": 1175
      -  "k1176": 1176
      -  "k1177": 1177
      -  "k1178": 1178
      -  "k1179": 1179
      -  "k118": 118
      -  "k1180": 1180
      -  "k1181": 1181
      -  "k1182": 1182
      -  "k1183": 1183
      -  "k1184": 1184
      -  "k1185": 1185
      -  "k1186": 1186
      -  "k1187": 1187
      -  "k1188": 1188
      -  "k1189": 1189
      -  "k119": 119
      -  "k1190": 1190
      -  "k1191": 1191
      -  "k1192": 1192
      -  "k1193": 1193
      -  "k1194": 1194
      -  "k1195": 1195
      -  "k1196": 1196
      -  "k1197": 1197
      -  "k1198": 1198
      -  "k1199": 1199
      -  "k12": 12
      -  "k120": 120
      -  "k1200": 1200
      -  "k1201": 1201
      -  "k1202": 1202
      -  "k1203": 1203
      -  "k1204": 1204
      -  "k1205": 1205
      -  "k1206": 1206
      -  "k1207": 1207
      -  "k1208": 1208
      -  "k1209": 1209
      -  "k121": 121
      -  "k1210": 1210
      -  "k1211": 1211
      -  "k1212": 1212
      -  "k1213": 1213
      -  "k1214": 1214
      -  "k1215": 1215
      -  "k1216": 1216
      -  "k1217": 1217
      -  "k1218": 1218
      -  "k1219": 1219
      -  "k122": 122
      -  "k1220": 1220
      -  "k1221": 1221
      -  "k1222": 1222
      -  "k1223": 1223
      -  "k1224": 1224
      -  "k1225": 1225
      -  "k1226": 1226
      -  "k1227": 1227
      -  "k1228": 1228
      -  "k1229": 1229
      -  "k123": 123
      -  "k1230": 1230
      -  "k1231": 1231
      -  "k1232": 1232
      -  "k1233": 1233
      -  "k1234": 1234
      -  "k1235": 1235
      -  "k1236": 1236
      -  "k1237": 1237
      -  "k1238": 1238
      -  "k1239": 1239
      -  "k124": 124
      -  "k1240": 1240
      -  "k1241": 1241
      -  "k1242": 1242
      -  "k1243": 1243
      -  "k1244": 1244
      -  "k1245": 1245
      -  "k1246": 1246
      -  "k1247": 1247
      -  "k1248": 1248
      -  "k1249": 1249
      -  "k125": 125
      -  "k1250": 1250
      -  "k1251": 1251
      -  "k1252": 1252
      -  "k1253": 1253
      -  "k1254": 1254
      -  "k1255": 1255
      -  "k1256": 1256
      -  "k1257": 1257
      -  "k1258": 1258
      -  "k1259": 1259
      -  "k126": 126
      -  "k1260": 1260
      -  "k1261": 1261
      -  "k1262": 1262
      -  "k1263": 1263
      -  "k1264": 1264
      -  "k1265": 1265
      -  "k1266": 1266
      -  "k1267": 1267
      -  "k1268": 1268
      -  "k1269": 1269
      -  "k127": 127
      -  "k1270": 1270
      -  "k1271": 1271
      -  "k1272": 1272
      -  "k1273": 1273
      -  "k1274": 1274
      -  "k1275": 1275
      -  "k1276": 1276
      -  "k1277": 1277
      -  "k1278": 1278
      -  "k1279": 1279
      -  "k128": 128
      -  "k1280": 1280
      -  "k1281": 1281
      -  "k1282": 1282
      -  "k1283": 1283
      -  "k1284": 1284
      -  "k1285": 1285
      -  "k1286": 1286
      -  "k1287": 1287
      -  "k1288": 1288
      -  "k1289": 1289
      -  "k129": 129
      -  "k1290": 1290
      -  "k1291": 1291
      -  "k1292": 1292
      -  "k1293": 1293
      -  "k1294": 1294
      -  "k1295": 1295
      -  "k1296": 1296
      -  "k1297": 1297
      -  "k1298": 1298
      -  "k1299": 1299
      -  "k13": 13
      -  "k130": 130
      -  "k1300": 1300
      -  "k1301": 1301
      -  "k1302": 1302
      -  "k1303": 1303
      -  "k1304": 1304
      -  "k1305": 1305
      -  "k1306": 1306
      -  "k1307": 1307
      -  "k1308": 1308
      -  "k1309": 1309
      -  "k131": 131
      -  "k1310": 1310
      -  "k1311": 1311
      -  "k1312": 1312
      -  "k1313": 1313
      -  "k1314": 1314
      -  "k1315": 1315
      -  "k1316": 1316
      -  "k1317": 1317
      -  "k1318": 1318
      -  "k1319": 1319
      -  "k132": 132
      -  "k1320": 1320
      -  "k1321": 1321
      -  "k1322": 1322
      -  "k1323": 1323
      -  "k1324": 1324
      -  "k1325": 1325
      -  "k1326": 1326
      -  "k1327": 1327
      -  "k1328": 1328
      -  "k1329": 1329
      -  "k133": 133
      -  "k1330": 1330
      -  "k1331": 1331
      -  "k1332": 1332
      -  "k1333": 1333
      -  "k1334": 1334
      -  "k1335": 1335
      -  "k1336": 1336
      -  "k1337": 1337
      -  "k1338": 1338
      -  "k1339": 1339
      -  "k134": 134
      -  "k1340": 1340
      -  "k1341": 1341
      -  "k1342": 1342
      -  "k1343": 1343
      -  "k1344": 1344
      -  "k1345": 1345
      -  "k1346": 1346
      -  "k1347": 1347
      -  "k1348": 1348
      -  "k1349": 1349
      -  "k135": 135
      -  "k1350": 1350
      -  "k1351": 1351
      -  "k1352": 1352
      -  "k1353": 1353
      -  "k1354": 1354
      -  "k1355": 1355
      -  "k1356": 1356
      -  "k1357": 1357
      -  "k1358": 1358
      -  "k1359": 1359
      -  "k136": 136
      -  "k1360": 1360
      -  "k1361": 1361
      -  "k1362": 1362
      -  "k1363": 1363
      -  "k1364": 1364
      -  "k1365": 1365
      -  "k1366": 1366
      -  "k1367": 1367
      -  "k1368": 1368
      -  "k1369": 1369
      -  "k137": 137
      -  "k1370": 1370
      -  "k1371": 1371
      -  "k1372": 1372
      -  "k1373": 1373
      -  "k1374": 1374
      -  "k1375": 1375
      -  "k1376": 1376
      -  "k1377": 1377
      -  "k1378": 1378
      -  "k1379": 1379
      -  "k138": 138
      -  "k1380": 1380
      -  "k1381": 1381
      -  "k1382": 1382
      -  "k1383": 1383
      -  "k1384": 1384
      -  "k1385": 1385
      -  "k1386": 1386
      -  "k1387": 1387
      -  "k1388": 1388
      -  "k1389": 1389
      -  "k139": 139
      -  "k1390": 1390
      -  "k1391": 1391
      -  "k1392": 1392
      -  "k1393": 1393
      -  "k1394": 1394
      -  "k1395": 1395
      -  "k1396": 1396
      -  "k1397": 1397
      -  "k1398": 1398
      -  "k1399": 1399
      -  "k14": 14
      -  "k140": 140
      -  "k1400": 1400
      -  "k1401": 1401
      -  "k1402": 1402
      -  "k1403": 1403
      -  "k1404": 1404
      -  "k1405": 1405
      -  "k1406": 1406
      -  "k1407": 1407
      -  "k1408": 1408
      -  "k1409": 1409
      -  "k141": 141
      -  "k1410": 1410
      -  "k1411": 1411
      -  "k1412": 1412
      -  "k1413": 1413
      -  "k1414": 1414
      -  "k1415": 1415
      -  "k1416": 1416
      -  "k1417": 1417
      -  "k1418": 1418
      -  "k1419": 1419
      -  "k142": 142
      -  "k1420": 1420
      -  "k1421": 1421
      -  "k1422": 1422
      -  "k1423": 1423
      -  "k1424": 1424
      -  "k1425": 1425
      -  "k1426": 1426
      -  "k1427": 1427
      -  "k1428": 1428
      -  "k1429": 1429
      -  "k143": 143
      -  "k1430": 1430
      -  "k1431": 1431
      -  "k1432": 1432
      -  "k1433": 1433
      -  "k1434": 1434
      -  "k1435": 1435
      -  "k1436": 1436
      -  "k1437": 1437
      -  "k1438": 1438
      -  "k1439": 1439
      -  "k144": 144
      -  "k1440": 1440
      -  "k1441": 1441
      -  "k1442": 1442
      -  "k1443": 1443
      -  "k1444": 1444
      -  "k1445": 1445
      -  "k1446": 1446
      -  "k1447": 1447
      -  "k1448": 1448
      -  "k1449": 1449
      -  "k145": 145
      -  "k1450": 1450
      -  "k1451": 1451
      -  "k1452": 1452
      -  "k1453": 1453
      -  "k1454": 1454
      -  "k1455": 1455
      -  "k1456": 1456
      -  "k1457": 1457
      -  "k1458": 1458
      -  "k1459": 1459
      -  "k146": 146
      -  "k1460": 1460
      -  "k1461": 1461
      -  "k1462": 1462
      -  "k1463": 1463
      -  "k1464": 1464
      -  "k1465": 1
      +{}
      
      [mocha] output truncated to 8192 characters, see "maxDiffSize" reporter-option

      at Context.<anonymous> (repro.test.js:32:20)
      at process.processImmediate (node:internal/timers:504:21)
```

</details>

--- a/lib/reporters/base.js
+++ b/lib/reporters/base.js
@@ -275,8 +275,73 @@ var showDiff = (exports.showDiff = function (err) {
   );
 });
 
+/**
+ * Estimate the serialized size of a value.
+ * Returns -1 if the value is too complex to estimate safely.
+ *
+ * @private
+ * @param {*} val
+ * @param {number} maxDepth
+ * @param {Set<*>} [seen]
+ * @returns {number}
+ */
+function estimateSize(val, maxDepth, seen) {
+  if (maxDepth <= 0) return -1;
+  
+  seen = seen || new Set();
+  
+  const type = typeof val;
+  if (type === 'string') return val.length;
+  if (type === 'number' || type === 'boolean') return 8;
+  if (val === null || val === undefined) return 4;
+  
+  // Avoid circular references
+  if (type === 'object') {
+    if (seen.has(val)) return -1;
+    seen.add(val);
+    
+    // Special handling for Buffers - they can be massive
+    if (typeof Buffer !== 'undefined' && Buffer.isBuffer(val)) {
+      return val.length > 10000 ? -1 : val.length * 2;
+    }
+    
+    let size = 2; // {} or []
+    try {
+      const keys = Object.keys(val);
+      if (keys.length > 1000) return -1; // Too many keys
+      
+      for (const key of keys) {
+        const childSize = estimateSize(val[key], maxDepth - 1, seen);
+        if (childSize === -1) return -1;
+        size += key.length + childSize + 4; // key + value + formatting
+        if (size > 100000) return -1; // Bail early if getting huge
+      }
+    } catch {
+      return -1;
+    }
+    
+    seen.delete(val);
+    return size;
+  }
+  
+  return 50; // Default estimate for other types
+}
+
 function stringifyDiffObjs(err) {
   if (!utils.isString(err.actual) || !utils.isString(err.expected)) {
+    // Estimate size before stringifying to avoid hangs
+    const maxSafeSize = exports.maxDiffSize || 8192;
+    const actualSize = estimateSize(err.actual, 10);
+    const expectedSize = estimateSize(err.expected, 10);
+    
+    if (actualSize === -1 || expectedSize === -1 || 
+        actualSize > maxSafeSize || expectedSize > maxSafeSize) {
+      // Values too large/complex - provide safe fallback
+      err.actual = '[object too large to diff]';
+      err.expected = '[object too large to diff]';
+      return;
+    }
+    
     err.actual = utils.stringify(err.actual);
     err.expected = utils.stringify(err.expected);
   }

--- a/lib/reporters/base.js
+++ b/lib/reporters/base.js
@@ -287,29 +287,29 @@ var showDiff = (exports.showDiff = function (err) {
  */
 function estimateSize(val, maxDepth, seen) {
   if (maxDepth <= 0) return -1;
-  
+
   seen = seen || new Set();
-  
+
   const type = typeof val;
   if (type === 'string') return val.length;
   if (type === 'number' || type === 'boolean') return 8;
   if (val === null || val === undefined) return 4;
-  
+
   // Avoid circular references
   if (type === 'object') {
     if (seen.has(val)) return -1;
     seen.add(val);
-    
+
     // Special handling for Buffers - they can be massive
     if (typeof Buffer !== 'undefined' && Buffer.isBuffer(val)) {
       return val.length > 10000 ? -1 : val.length * 2;
     }
-    
+
     let size = 2; // {} or []
     try {
       const keys = Object.keys(val);
       if (keys.length > 1000) return -1; // Too many keys
-      
+
       for (const key of keys) {
         const childSize = estimateSize(val[key], maxDepth - 1, seen);
         if (childSize === -1) return -1;
@@ -319,11 +319,11 @@ function estimateSize(val, maxDepth, seen) {
     } catch {
       return -1;
     }
-    
+
     seen.delete(val);
     return size;
   }
-  
+
   return 50; // Default estimate for other types
 }
 
@@ -333,15 +333,15 @@ function stringifyDiffObjs(err) {
     const maxSafeSize = exports.maxDiffSize || 8192;
     const actualSize = estimateSize(err.actual, 10);
     const expectedSize = estimateSize(err.expected, 10);
-    
-    if (actualSize === -1 || expectedSize === -1 || 
-        actualSize > maxSafeSize || expectedSize > maxSafeSize) {
+
+    if (actualSize === -1 || expectedSize === -1 ||
+      actualSize > maxSafeSize || expectedSize > maxSafeSize) {
       // Values too large/complex - provide safe fallback
       err.actual = '[object too large to diff]';
       err.expected = '[object too large to diff]';
       return;
     }
-    
+
     err.actual = utils.stringify(err.actual);
     err.expected = utils.stringify(err.expected);
   }

--- a/lib/reporters/base.js
+++ b/lib/reporters/base.js
@@ -291,17 +291,17 @@ function estimateSize(val, maxDepth, seen) {
   seen = seen || new Set();
 
   const type = typeof val;
-  if (type === 'string') return val.length;
-  if (type === 'number' || type === 'boolean') return 8;
+  if (type === "string") return val.length;
+  if (type === "number" || type === "boolean") return 8;
   if (val === null || val === undefined) return 4;
 
   // Avoid circular references
-  if (type === 'object') {
+  if (type === "object") {
     if (seen.has(val)) return -1;
     seen.add(val);
 
     // Special handling for Buffers - they can be massive
-    if (typeof Buffer !== 'undefined' && Buffer.isBuffer(val)) {
+    if (typeof Buffer !== "undefined" && Buffer.isBuffer(val)) {
       return val.length > 10000 ? -1 : val.length * 2;
     }
 
@@ -334,11 +334,15 @@ function stringifyDiffObjs(err) {
     const actualSize = estimateSize(err.actual, 10);
     const expectedSize = estimateSize(err.expected, 10);
 
-    if (actualSize === -1 || expectedSize === -1 ||
-      actualSize > maxSafeSize || expectedSize > maxSafeSize) {
+    if (
+      actualSize === -1 ||
+      expectedSize === -1 ||
+      actualSize > maxSafeSize ||
+      expectedSize > maxSafeSize
+    ) {
       // Values too large/complex - provide safe fallback
-      err.actual = '[object too large to diff]';
-      err.expected = '[object too large to diff]';
+      err.actual = "[object too large to diff]";
+      err.expected = "[object too large to diff]";
       return;
     }
 

--- a/test/integration/fixtures/diff-hang-protection.fixture.js
+++ b/test/integration/fixtures/diff-hang-protection.fixture.js
@@ -1,0 +1,41 @@
+'use strict';
+
+const {expect} = require('chai');
+
+describe('Diff hang protection', function () {
+  it('should not hang when comparing large buffers', function () {
+    // Create two large buffers that differ
+    const buf1 = Buffer.alloc(50000, 'a');
+    const buf2 = Buffer.alloc(50000, 'b');
+    
+    // This should fail without hanging
+    expect(buf1).to.deep.equal(buf2);
+  });
+  
+  it('should not hang when comparing deeply nested objects', function () {
+    // Create deeply nested object structure
+    let obj1 = {value: 1};
+    let obj2 = {value: 2};
+    
+    for (let i = 0; i < 100; i++) {
+      obj1 = {nested: obj1, extra: obj1};
+      obj2 = {nested: obj2, extra: obj2};
+    }
+    
+    // This should fail without hanging
+    expect(obj1).to.deep.equal(obj2);
+  });
+  
+  it('should not hang when comparing objects with many keys', function () {
+    const obj1 = {};
+    const obj2 = {};
+    
+    for (let i = 0; i < 5000; i++) {
+      obj1['key' + i] = 'value' + i;
+      obj2['key' + i] = 'different' + i;
+    }
+    
+    // This should fail without hanging
+    expect(obj1).to.deep.equal(obj2);
+  });
+});

--- a/test/integration/fixtures/diff-hang-protection.fixture.js
+++ b/test/integration/fixtures/diff-hang-protection.fixture.js
@@ -7,7 +7,6 @@ describe('Diff hang protection', function () {
     const buf1 = Buffer.alloc(50000, 'a');
     const buf2 = Buffer.alloc(50000, 'b');
     
-    // This should fail without hanging
     expect(buf1).to.deep.equal(buf2);
   });
   

--- a/test/integration/fixtures/diff-hang-protection.fixture.js
+++ b/test/integration/fixtures/diff-hang-protection.fixture.js
@@ -1,39 +1,36 @@
 'use strict';
 
-const {expect} = require('chai');
+const { expect } = require('chai');
 
 describe('Diff hang protection', function () {
   it('should not hang when comparing large buffers', function () {
     const buf1 = Buffer.alloc(50000, 'a');
     const buf2 = Buffer.alloc(50000, 'b');
-    
+
     expect(buf1).to.deep.equal(buf2);
   });
-  
+
   it('should not hang when comparing deeply nested objects', function () {
-    // Create deeply nested object structure
-    let obj1 = {value: 1};
-    let obj2 = {value: 2};
-    
+    let obj1 = { value: 1 };
+    let obj2 = { value: 2 };
+
     for (let i = 0; i < 100; i++) {
-      obj1 = {nested: obj1, extra: obj1};
-      obj2 = {nested: obj2, extra: obj2};
+      obj1 = { nested: obj1, extra: obj1 };
+      obj2 = { nested: obj2, extra: obj2 };
     }
-    
-    // This should fail without hanging
+
     expect(obj1).to.deep.equal(obj2);
   });
-  
+
   it('should not hang when comparing objects with many keys', function () {
     const obj1 = {};
     const obj2 = {};
-    
+
     for (let i = 0; i < 5000; i++) {
       obj1['key' + i] = 'value' + i;
       obj2['key' + i] = 'different' + i;
     }
-    
-    // This should fail without hanging
+
     expect(obj1).to.deep.equal(obj2);
   });
 });

--- a/test/integration/fixtures/diff-hang-protection.fixture.js
+++ b/test/integration/fixtures/diff-hang-protection.fixture.js
@@ -4,7 +4,6 @@ const {expect} = require('chai');
 
 describe('Diff hang protection', function () {
   it('should not hang when comparing large buffers', function () {
-    // Create two large buffers that differ
     const buf1 = Buffer.alloc(50000, 'a');
     const buf2 = Buffer.alloc(50000, 'b');
     

--- a/test/unit/reporters/base-diff-protection.spec.js
+++ b/test/unit/reporters/base-diff-protection.spec.js
@@ -1,0 +1,121 @@
+'use strict';
+
+var sinon = require('sinon');
+var reporters = require('../../../').reporters;
+var Base = reporters.Base;
+var chaiExpect = require('chai').expect;
+
+describe('Base reporter diff hang protection', function () {
+  var stdout;
+
+  var gather = function (chunk) {
+    stdout.push(chunk);
+  };
+
+  beforeEach(function () {
+    sinon.stub(Base, 'useColors').value(false);
+    sinon.stub(process.stdout, 'write').callsFake(gather);
+    stdout = [];
+  });
+
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  function list(tests) {
+    try {
+      Base.list(tests);
+    } finally {
+      sinon.restore();
+    }
+  }
+
+  function makeFailedTest(title, actual, expected) {
+    var err = new Error('values differ');
+    err.actual = actual;
+    err.expected = expected;
+    err.showDiff = true;
+    return {
+      isPending: function () { return false; },
+      titlePath: function () { return [title]; },
+      err: err,
+      fullTitle: function () { return title; },
+    };
+  }
+
+  describe('large buffers', function () {
+    it('should not hang when diffing large buffers', function () {
+      this.timeout(2000);
+
+      var buf1 = Buffer.alloc(50000, 'a');
+      var buf2 = Buffer.alloc(50000, 'b');
+      var test = makeFailedTest('large buffer test', buf1, buf2);
+
+      var start = Date.now();
+      list([test]);
+      var elapsed = Date.now() - start;
+
+      chaiExpect(elapsed).to.be.below(1000);
+      // Verify the err was handled (either truncated or replaced with fallback)
+      chaiExpect(test.err.actual).to.be.a('string');
+    });
+  });
+
+  describe('deeply nested objects', function () {
+    it('should not hang when diffing deeply nested objects', function () {
+      this.timeout(2000);
+
+      var obj1 = {value: 1};
+      var obj2 = {value: 2};
+
+      for (var i = 0; i < 50; i++) {
+        obj1 = {nested: obj1, extra: obj1};
+        obj2 = {nested: obj2, extra: obj2};
+      }
+
+      var test = makeFailedTest('nested object test', obj1, obj2);
+
+      var start = Date.now();
+      list([test]);
+      var elapsed = Date.now() - start;
+
+      chaiExpect(elapsed).to.be.below(1000);
+      chaiExpect(test.err.actual).to.be.a('string');
+    });
+  });
+
+  describe('objects with many keys', function () {
+    it('should not hang when diffing objects with many keys', function () {
+      this.timeout(2000);
+
+      var obj1 = {};
+      var obj2 = {};
+
+      for (var i = 0; i < 2000; i++) {
+        obj1['key' + i] = 'value' + i;
+        obj2['key' + i] = 'different' + i;
+      }
+
+      var test = makeFailedTest('many keys test', obj1, obj2);
+
+      var start = Date.now();
+      list([test]);
+      var elapsed = Date.now() - start;
+
+      chaiExpect(elapsed).to.be.below(1000);
+      chaiExpect(test.err.actual).to.be.a('string');
+    });
+  });
+
+  describe('normal-sized objects', function () {
+    it('should still produce a valid diff for small objects', function () {
+      var test = makeFailedTest('small object test', {a: 1, b: 2}, {a: 1, b: 3});
+
+      list([test]);
+
+      var output = stdout.join('');
+      // The diff should contain the actual values, not the fallback message
+      chaiExpect(output).to.not.contain('too large to diff');
+    });
+  });
+});

--- a/test/unit/reporters/base-diff-protection.spec.js
+++ b/test/unit/reporters/base-diff-protection.spec.js
@@ -1,11 +1,11 @@
-'use strict';
+"use strict";
 
-var sinon = require('sinon');
-var reporters = require('../../../').reporters;
+var sinon = require("sinon");
+var reporters = require("../../../").reporters;
 var Base = reporters.Base;
-var chaiExpect = require('chai').expect;
+var chaiExpect = require("chai").expect;
 
-describe('Base reporter diff hang protection', function () {
+describe("Base reporter diff hang protection", function () {
   var stdout;
 
   var gather = function (chunk) {
@@ -13,8 +13,8 @@ describe('Base reporter diff hang protection', function () {
   };
 
   beforeEach(function () {
-    sinon.stub(Base, 'useColors').value(false);
-    sinon.stub(process.stdout, 'write').callsFake(gather);
+    sinon.stub(Base, "useColors").value(false);
+    sinon.stub(process.stdout, "write").callsFake(gather);
     stdout = [];
   });
 
@@ -31,25 +31,31 @@ describe('Base reporter diff hang protection', function () {
   }
 
   function makeFailedTest(title, actual, expected) {
-    var err = new Error('values differ');
+    var err = new Error("values differ");
     err.actual = actual;
     err.expected = expected;
     err.showDiff = true;
     return {
-      isPending: function () { return false; },
-      titlePath: function () { return [title]; },
+      isPending: function () {
+        return false;
+      },
+      titlePath: function () {
+        return [title];
+      },
       err: err,
-      fullTitle: function () { return title; },
+      fullTitle: function () {
+        return title;
+      },
     };
   }
 
-  describe('large buffers', function () {
-    it('should not hang when diffing large buffers', function () {
+  describe("large buffers", function () {
+    it("should not hang when diffing large buffers", function () {
       this.timeout(2000);
 
-      var buf1 = Buffer.alloc(50000, 'a');
-      var buf2 = Buffer.alloc(50000, 'b');
-      var test = makeFailedTest('large buffer test', buf1, buf2);
+      var buf1 = Buffer.alloc(50000, "a");
+      var buf2 = Buffer.alloc(50000, "b");
+      var test = makeFailedTest("large buffer test", buf1, buf2);
 
       var start = Date.now();
       list([test]);
@@ -57,12 +63,12 @@ describe('Base reporter diff hang protection', function () {
 
       chaiExpect(elapsed).to.be.below(1000);
       // Verify the err was handled (either truncated or replaced with fallback)
-      chaiExpect(test.err.actual).to.be.a('string');
+      chaiExpect(test.err.actual).to.be.a("string");
     });
   });
 
-  describe('deeply nested objects', function () {
-    it('should not hang when diffing deeply nested objects', function () {
+  describe("deeply nested objects", function () {
+    it("should not hang when diffing deeply nested objects", function () {
       this.timeout(2000);
 
       var obj1 = { value: 1 };
@@ -73,48 +79,52 @@ describe('Base reporter diff hang protection', function () {
         obj2 = { nested: obj2, extra: obj2 };
       }
 
-      var test = makeFailedTest('nested object test', obj1, obj2);
+      var test = makeFailedTest("nested object test", obj1, obj2);
 
       var start = Date.now();
       list([test]);
       var elapsed = Date.now() - start;
 
       chaiExpect(elapsed).to.be.below(1000);
-      chaiExpect(test.err.actual).to.be.a('string');
+      chaiExpect(test.err.actual).to.be.a("string");
     });
   });
 
-  describe('objects with many keys', function () {
-    it('should not hang when diffing objects with many keys', function () {
+  describe("objects with many keys", function () {
+    it("should not hang when diffing objects with many keys", function () {
       this.timeout(2000);
 
       var obj1 = {};
       var obj2 = {};
 
       for (var i = 0; i < 2000; i++) {
-        obj1['key' + i] = 'value' + i;
-        obj2['key' + i] = 'different' + i;
+        obj1["key" + i] = "value" + i;
+        obj2["key" + i] = "different" + i;
       }
 
-      var test = makeFailedTest('many keys test', obj1, obj2);
+      var test = makeFailedTest("many keys test", obj1, obj2);
 
       var start = Date.now();
       list([test]);
       var elapsed = Date.now() - start;
 
       chaiExpect(elapsed).to.be.below(1000);
-      chaiExpect(test.err.actual).to.be.a('string');
+      chaiExpect(test.err.actual).to.be.a("string");
     });
   });
 
-  describe('normal-sized objects', function () {
-    it('should still produce a valid diff for small objects', function () {
-      var test = makeFailedTest('small object test', { a: 1, b: 2 }, { a: 1, b: 3 });
+  describe("normal-sized objects", function () {
+    it("should still produce a valid diff for small objects", function () {
+      var test = makeFailedTest(
+        "small object test",
+        { a: 1, b: 2 },
+        { a: 1, b: 3 },
+      );
 
       list([test]);
 
-      var output = stdout.join('');
-      chaiExpect(output).to.not.contain('too large to diff');
+      var output = stdout.join("");
+      chaiExpect(output).to.not.contain("too large to diff");
     });
   });
 });

--- a/test/unit/reporters/base-diff-protection.spec.js
+++ b/test/unit/reporters/base-diff-protection.spec.js
@@ -65,12 +65,12 @@ describe('Base reporter diff hang protection', function () {
     it('should not hang when diffing deeply nested objects', function () {
       this.timeout(2000);
 
-      var obj1 = {value: 1};
-      var obj2 = {value: 2};
+      var obj1 = { value: 1 };
+      var obj2 = { value: 2 };
 
       for (var i = 0; i < 50; i++) {
-        obj1 = {nested: obj1, extra: obj1};
-        obj2 = {nested: obj2, extra: obj2};
+        obj1 = { nested: obj1, extra: obj1 };
+        obj2 = { nested: obj2, extra: obj2 };
       }
 
       var test = makeFailedTest('nested object test', obj1, obj2);
@@ -109,12 +109,11 @@ describe('Base reporter diff hang protection', function () {
 
   describe('normal-sized objects', function () {
     it('should still produce a valid diff for small objects', function () {
-      var test = makeFailedTest('small object test', {a: 1, b: 2}, {a: 1, b: 3});
+      var test = makeFailedTest('small object test', { a: 1, b: 2 }, { a: 1, b: 3 });
 
       list([test]);
 
       var output = stdout.join('');
-      // The diff should contain the actual values, not the fallback message
       chaiExpect(output).to.not.contain('too large to diff');
     });
   });

